test.py

```diff
@@ -0,0 +1,122 @@
+from objects.factor import Factor
+from functions import sumout,multiply
+from objects.key_iterator import KeyIterator
+from objects.relation import Relation
+
+relation1 = Relation('A|B')
+relation2 = Relation('A,B')
+relation3 = Relation('A|B,C', ['-b'])
+relation4 = Relation('A,B,C', ['-c'])
+
+assert(relation1.is_conditional)
+assert(relation2.is_joint)
+assert(relation3.is_conditional)
+assert(relation4.is_joint)
+
+print(relation1.variables, relation1.query_variables, relation1.evidence_variables)
+print(relation2.variables)
+print(relation3.variables, relation3.query_variables, relation3.evidence_variables)
+print(relation4.variables)
+print(relation3.values)
+print(relation4.values)
+
+kit1 = KeyIterator(relation1)
+kit1_rows = 0
+for row in kit1:
+    kit1_rows += 1
+print(kit1)
+assert(kit1_rows == 4)
+
+kit2 = KeyIterator(relation2)
+kit2_rows = 0
+for row in kit2:
+    kit2_rows += 1
+print(kit2)
+assert(kit2_rows == 4)
+
+
+kit3 = KeyIterator(relation3)
+kit3_rows = 0
+for row in kit3:
+    kit3_rows += 1
+print(kit3)
+assert(kit3_rows == 4)
+
+kit4 = KeyIterator(relation4)
+kit4_rows = 0
+for row in kit4:
+    kit4_rows += 1
+print(kit4)
+assert(kit4_rows == 4)
+
+
+# Rain-traffic-late example given +r, P(+r), P(T|+r), P(L|T)
+rain2 = Factor('R', ['+r'])
+rain2.init([0.1])
+print(rain2)
```

Comment:
For this print and below ones too, probably better as asserts with expected values